connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -154,7 +154,7 @@ public void onShutdown(String connector) {
 
     @Override
     public void onFailure(String connector, Throwable cause) {
-        statusBackingStore.putSafe(new ConnectorStatus(connector, ConnectorStatus.State.FAILED,
+        statusBackingStore.put(new ConnectorStatus(connector, ConnectorStatus.State.FAILED,
```

Comment:
Interesting. I think the case I was concerned about when creating the poorly named `putSafe` methods was a defunct task which didn't shutdown quick enough and then ultimately transitioned itself to FAILED after the group had already rebalanced. Maybe the `cancelled` flag provides enough protection from this case? Should we have something similar in `WorkerConnector`?